velox/expression/tests/ExpressionFuzzer.h

```diff
@@ -337,41 +319,61 @@ class ExpressionFuzzer {
   // --assign_function_tickets startup flag .
   std::unordered_map<std::string, int> functionsToTickets_;
 
-  /// The remaining levels of expression nesting. It's initialized by
-  /// FLAGS_max_level_of_nesting and updated in generateExpression(). When its
-  /// value decreases to 0, we don't generate subexpressions anymore.
-  int32_t remainingLevelOfNesting_;
-
   /// We allow the arg generation routine to be specialized for particular
   /// functions. This map stores the mapping between function name and the
   /// overridden method.
   using ArgsOverrideFunc = std::function<std::vector<core::TypedExprPtr>(
       const CallableSignature& input)>;
+
   std::unordered_map<std::string, ArgsOverrideFunc> funcArgOverrides_;
 
-  std::shared_ptr<core::QueryCtx> queryCtx_{std::make_shared<core::QueryCtx>()};
-  std::shared_ptr<memory::MemoryPool> pool_{memory::addDefaultLeafMemoryPool()};
-  core::ExecCtx execCtx_{pool_.get(), queryCtx_.get()};
-  test::ExpressionVerifier verifier_;
+  // This pool is needed to generate vectors when constant expressions get
+  // generated.
+  std::shared_ptr<memory::MemoryPool> pool_;
 
-  test::VectorMaker vectorMaker_{execCtx_.pool()};
   VectorFuzzer vectorFuzzer_;
 
+  FuzzerGenerator rng_;
+
   /// Contains the input column references that need to be generated for one
   /// particular iteration.
-  std::vector<TypePtr> inputRowTypes_;
-  std::vector<std::string> inputRowNames_;
+  struct State {
+    void reset() {
```

Comment:
Why doesn't reset() change remainingLevelOfNesting_ too? By the way, I think remainingLevelOfNesting_ is not a per-iteration, but per-expression state.

Reply:
reset here is just restting what was reset originally in the original code. 
note that remainingLevelOfNesting_ is being updated  up and down automatically 
so when reset is called it should always be 0. 
I can add a reset here. but this did not exist in the original code. 
```
core::TypedExprPtr ExpressionFuzzer::generateExpression(
    const TypePtr& returnType) {
  VELOX_CHECK_GT(state.remainingLevelOfNesting_, 0);
  --state.remainingLevelOfNesting_;
  auto guard = folly::makeGuard([&] { ++state.remainingLevelOfNesting_; });

  core::TypedExprPtr expression;
  ```